.github/workflows/publish.yml

```
@@ -6,6 +6,16 @@ on:
     tags: ['*']
 
 jobs:
+  version-check:
+    # We need this job to run only on push with tag.
+    if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
```

There is no other event types:
```suggestion
    if: startsWith(github.ref, 'refs/tags/')
```

---

[Delete](https://docs.github.com/en/webhooks-and-events/events/github-event-types#deleteevent)

---

It does not trigger on `git push --delete origin tag`

https://github.com/oleg-jukovec/expirationd/actions

---

The reason is than workflow itself doesn't trigger on delete: https://github.com/oleg-jukovec/expirationd/blob/c872d7313eda324985212382fd1f3623f4fd44dd/.github/workflows/publish.yml#L3-L6

---

Yeah, but it the `publish.yml` we have the same:

https://github.com/tarantool/expirationd/blob/master/.github/workflows/publish.yml#L3-L7

---

https://github.com/tarantool/etcd-client/pull/30#discussion_r1138557760